File: MFrontend/MFrontend/Platforms/Android/MyPrintService.cs
using Android.App;
using Android.Content;
using Android.Net.Wifi.Aware;
using Android.PrintServices;
using Android.Util;
using AndroidX.Annotations;
using Newtonsoft.Json;


namespace MFrontend.Platforms.Android
{
    [Service(
        Exported = true,
        Permission = "android.permission.BIND_PRINT_SERVICE"
    )]
    [IntentFilter(new[] { "android.printservice.PrintService" })]
    public class MyPrintService : PrintService
    {
        const string TAG = "MyPrintService";

        public BackendPrinter[] SharedPrinters = new BackendPrinter[0];
        //private MyPrinterDiscoverySession? _discoverySession;

        public MyPrintService()
        {

        }

        public BackendPrinter[] GetCurrentPrinters() => SharedPrinters ?? Array.Empty<BackendPrinter>();


        protected override PrinterDiscoverySession OnCreatePrinterDiscoverySession()
        {
            var session = new MyPrinterDiscoverySession(this, SharedPrinters);
            //_discoverySession = session;
            return session;
        }

        protected override void OnRequestCancelPrintJob(PrintJob printJob)
        {
            Log.Debug(TAG, $"Cancelling print job: {printJob.Id?.ToString()}");
            printJob.Cancel();
        }

        protected override void OnPrintJobQueued(PrintJob printJob)
        {
            Log.Debug(TAG, $"Print job queued: {printJob.Id?.ToString()}");
            printJob.Start();
            printJob.Complete();
        }

        //public void PrintersToDiscover(BackendPrinter[] printers)
        //{
        //    SharedPrinters = printers;
        //    Log.Debug(TAG, $"PrintersToDiscover called with {printers.Length} printers");
        //}

        public override void OnCreate()
        {
            base.OnCreate();
            var prefs = ApplicationContext
                .GetSharedPreferences("print_data", FileCreationMode.Private);
            var json = prefs.GetString("stored_printers", "[]");
            SharedPrinters = JsonConvert.DeserializeObject<BackendPrinter[]>(json)
                             ?? Array.Empty<BackendPrinter>();
            Log.Debug("MyPrintService", $"Loaded {SharedPrinters.Length} printers from SharedPreferences");
        }
    }
}

File: FFrontend/stage_uitvoeringen/android/app/src/main/java/com/example/stage_uitvoeringen/MyPrintService.java
package com.example.stage_uitvoeringen;

import android.content.Context;
import android.content.SharedPreferences;
import android.printservice.PrintJob;
import android.printservice.PrintService;
import android.printservice.PrinterDiscoverySession;
import android.util.Log;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;

import androidx.annotation.Nullable;

public class MyPrintService extends PrintService {
    public static BackendPrinter[] sharedPrinters = new BackendPrinter[0];
    private static final String TAG = "MyPrintServiceFlutter";

    public static void printersToDiscover(BackendPrinter[] printers){
        sharedPrinters = printers;
    }

    @Nullable
    @Override
    protected PrinterDiscoverySession onCreatePrinterDiscoverySession() {
        Log.d(TAG, "Creating new printer discovery session");
        return new MyPrinterDiscoverySession(this, sharedPrinters);
    }

    @Override
    protected void onRequestCancelPrintJob(PrintJob printJob) {
        Log.d(TAG, "Cancelling print job: " + printJob.getId());
        printJob.cancel();
    }

    @Override
    protected void onPrintJobQueued(PrintJob printJob) {
        Log.d(TAG, "Print job queued: " + printJob.getId());
        printJob.start();
        printJob.complete();
    }

    @Override
    public void onCreate() {
        super.onCreate();

        // Retrieve the JSON string from SharedPreferences
        SharedPreferences prefs = getApplicationContext()
                .getSharedPreferences("print_data_flutter", Context.MODE_PRIVATE);
        String json = prefs.getString("stored_printers", "[]");

        // Deserialize JSON into BackendPrinter[]
        try {
            Gson gson = new Gson();
            BackendPrinter[] printersFromPrefs = gson.fromJson(json, BackendPrinter[].class);
            sharedPrinters = (printersFromPrefs != null)
                    ? printersFromPrefs
                    : new BackendPrinter[0];
        } catch (JsonSyntaxException e) {
            Log.e(TAG, "Failed to parse stored_printers JSON", e);
            sharedPrinters = new BackendPrinter[0];
        }

        Log.d(TAG, "Loaded " + sharedPrinters.length + " printers from SharedPreferences");
    }
}